### ISSUE TITLE
Support for Exponential Gradient Boosting

### ISSUE THREAD

**Issue by AnishShah** [pytorch/pytorch#2122]
[Be Careful What You Backpropagate: A Case For
Linear Output Activations & Gradient Boosting](https://arxiv.org/abs/1707.04199)

I can work on this if this can be added to pytorch. Please let me know. Thanks!

**Comment by jekbradbury** [comment]
This should be fairly straightforward to add in user code using `var.register_hook`. Eventually it could be added to core, but its utility should probably be validated in external repos first.
